cleanhttp.go

```diff
@@ -26,3 +29,11 @@ func DefaultClient() *http.Client {
 		Transport: DefaultTransport(),
 	}
 }
+
+func SetFinalizer(transport *http.Transport) {
+	runtime.SetFinalizer(&transport, FinalizeTransport)
+}
+
+func FinalizeTransport(t **http.Transport) {
+	(*t).CloseIdleConnections()
```

Comment:
Let's inline it. As we note "shipping is forever." especially with Go. So let's error on the side of smaller API surface, and only expose as needed, though I doubt it in this case.